### PR TITLE
Live game page

### PR DESCRIPTION
**Pull request by 3ricLu**
- Created all components in the live game page
- Lobby creation component that creates a lobby object
- Live game component that displays all lobby objects
- Created classes user, review, lobby, and game
- Created test data

### CODE REVIEW

File: client/src/models/User.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import background from "../ui/assets/images/background.png";
+
+export class User {
+  private id: string;
```
Comment (varangill) [inline client/src/models/User.tsx:4]: add underscores to private vars

File: client/src/models/User.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    auth: boolean = false,
+    friends: string[] = ["Novatic", "Rensuo"],
+    status: boolean = false,
+    introvert: number = 5,
+    observant: number = 5,
+    thinking: number = 5,
+    judging: number = 5
+  ) {
+    this.id = id || "000";
+    this.gameTags = gameTags || ["Valorant", "League of Legends"];
```
Comment (varangill) [inline client/src/models/User.tsx:34]: maybe mention removing defaults in a TODO

File: client/src/ui/components/Dropdown.tsx [hunk 3]
```diff
@@ -43,16 +50,33 @@ const Dropdown: React.FC<DropdownProps> = ({
     }
   };
 
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
```
Comment (varangill) [inline client/src/ui/components/Dropdown.tsx:56]: put boolean logic that isn't immediately clear in a const and use that const

so for example, do
const isUserClickingOutside = ...

File: client/src/ui/components/Dropdown.tsx [hunk 4]
```diff
@@ -43,16 +50,33 @@ const Dropdown: React.FC<DropdownProps> = ({
...
   return (
-    <div>
+    <div ref={dropdownRef} className="relative z-9">
       <button
         onClick={toggleDropdown}
         className="dropdownSearchButton justify-between btn-default h-9 flex rounded-xl bg-gray-100 hover:bg-gray-300 w-full items-center justify-center w-full"
       >
-        <div className="btn-text flex whitespace-nowrap overflow-hidden text-ellipsis w-11/12 px-4 text-white h-8 font-bold items-center text-xl sm:text-base">
+        <div className="btn-text flex whitespace-nowrap overflow-hidden text-ellipsis w-11/12 px-4 text-white h-8 font-bold items-center xl:text-md lg:text-md md:text-base sm:text-base">
           {selected.join(", ") || placeholder} {/* Show all selected items */}
```
Comment (varangill) [inline client/src/ui/components/Dropdown.tsx:80]: try `selected ? selected.join(", ) : placeholder`

File: client/src/ui/components/LiveGame.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    <div className="live-game-container w-full bg-gray-200 text-white rounded-2xl h-full flex flex-col">
+      <div className="font-bold w-full text-center items-center p-2 xl:text-2xl lg:text-2xl md:text-base sm:text-xl">
+        LIVE GAME
+      </div>
+      <div className="h-1 bg-gray-100 w-full"></div>
+      <div className="flex justify-between items-center m-2">
+        <div className="flex items-center gap-3 ml-4">
+          <button className="filter-button flex items-center mr-4">
+            <FontAwesomeIcon icon={faFilter} />
+            <i className="fas fa-filter mr-2"></i>FILTER
```
Comment (varangill) [inline client/src/ui/components/LiveGame.tsx:25]: close element inline 
`<i className="fas fa-filter mr-2" /i>`

**Review by varangill — APPROVED** [review]
couple nits

File: client/src/ui/components/Dropdown.tsx (hunk 3 above)
Comment (3ricLu) [reply client/src/ui/components/Dropdown.tsx:56]: image

Renamed constant

Comment (varangill) [reply client/src/ui/components/Dropdown.tsx:56]: const isUserClickingOutside =dropdownRef.current &&
      !dropdownRef.current.contains(event.target as Node

if (isUserClickingOutside ) {
setIsOpen(false)
}

**Pull request merged**